cmd/antrea-agent/agent.go

```diff
@@ -326,11 +329,21 @@ func run(o *Options) error {
 		l7NetworkPolicyEnabled,
 		l7FlowExporterEnabled,
 		o.config.DisableTXChecksumOffload)
-	if err := agentInitializer.Initialize(ctx); err != nil {
-		return fmt.Errorf("error initializing agent: %w", err)
+	err = agentInitializer.Initialize()
+	if err != nil {
+		return fmt.Errorf("error initializing agent: %v", err)
 	}
 	nodeConfig := agentInitializer.GetNodeConfig()
 
+	var nftablesSupported bool
+	if _, err := nftclient.New(networkConfig.IPv4Enabled, networkConfig.IPv6Enabled); err != nil {
+		klog.InfoS("nftables is not supported on this Node, skipping nftables-related features", "err", err)
+		nftablesSupported = false
+	} else {
+		klog.InfoS("nftables is supported on this Node")
+		nftablesSupported = true
+	}
+
```

Comment:
We can just move the logic L338-L346 to function `dumpNFTables` in dump_others.go, and we don't need the changes of introducing extra parameter `nftablesSupported`, simplifying the PR.